base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
I am not to sold on the current names but I think they should not be called `Record` + something. I initially thought it odd that rows and records were separate in https://github.com/willtim/row-polymorphism but in the end it makes sense as rows are a 'part' of a record but is not a record itself (it even has a different kind).